test/jdk/sun/net/www/http/KeepAliveCache/B5045306.java

```diff
@@ -183,20 +194,21 @@ public void request(HttpTransaction trans) {
                 System.out.println("server sleeping...");
                 try {Thread.sleep(6000); } catch (InterruptedException e) {}
 
-                trans.sendResponse(200, "OK");
+                trans.sendResponseHeaders(200, -1);
             } else if(path.equals("/part2")) {
                 System.out.println("Call to /part2");
                 byte[] responseBody = new byte[RESPONSE_DATA_LENGTH];
                 for (int i=0; i<responseBody.length; i++)
                     responseBody[i] = 0x41;
-                trans.setResponseEntityBody (responseBody, responseBody.length);
 
                 // override the Content-length header to be greater than the actual response body
-                trans.setResponseHeader("Content-length", Integer.toString(responseBody.length+1));
-                trans.sendResponse(200, "OK");
-
+                trans.getResponseHeaders().set("Content-length", Integer.toString(responseBody.length+1));
+                trans.sendResponseHeaders(200, 0);
```

Comment:
In case of HttpExchange.setResponseHeader(). If responseLength is -1, then content-length value is overridden to 0, if already set explicitly. Same is the case when responseLength is > 0. Only in the case when responseLength == 0, content-length value is not overriden if already set explicitly., that's why I am using chunked encoding and writing the data less than the content length.

Reply:
I understand why you do it - but the client will react differently if both Content-Length *and* chunk are specified, as opposed to when only Content-Length is specified. So I just want to make sure that we are testing the same thing than before. If we are not testing the same thing, then you might have to use a ServerSocket directly - rather than an HttpServer, to make sure we're sending back the same things than before.

Reply:
Now not using Transfer-Encoding Chunk.